tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2966,84 +2967,134 @@ llvm::Optional<Value> convertResizeOp(PatternRewriter& rewriter, Operation* op,
+  if (align_corners || uniform_sampling) {
+    int gcd_y = gcd(input_height - 1, output_height - 1);
+    int gcd_x = gcd(input_width - 1, output_width - 1);
+    scale_y_n = (output_height - 1) / gcd_y;
+    scale_y_d = (input_height - 1) / gcd_y;
+    scale_x_n = (output_width - 1) / gcd_x;
+    scale_x_d = (input_width - 1) / gcd_x;
+    offset_y = 0;
+    offset_x = 0;
+    border_y = 0;
+    border_x = 0;
+  }
+
+  if (!align_corners && !half_pixel_centers) {
+    // Adds a sampling offset.
+    offset_x += 1;
+    offset_y += 1;
+    // Adjust the borders to match an expected output shape.
+    border_x += 1;
+    border_y += 1;
+  }
+
+  ArrayAttr scale =
+      rewriter.getI64ArrayAttr({scale_y_n, scale_y_d, scale_x_n, scale_x_d});
+  ArrayAttr offset = rewriter.getI64ArrayAttr({offset_y, offset_x});
+  ArrayAttr border = rewriter.getI64ArrayAttr({border_y, border_x});
 
-  ArrayAttr output_size =
-      rewriter.getI64ArrayAttr({static_cast<int64_t>(output_height),
-                                static_cast<int64_t>(output_width)});
   StringAttr resize_mode = rewriter.getStringAttr(mode);
 
+  auto isInt16Range = [](int x) {
+    return (x <= std::numeric_limits<int16_t>::max()) &&
+           (x >= std::numeric_limits<int16_t>::min());
+  };
+
   if (input_is_qtype) {
-    // Magic shift number TFLite resize bilinear use
-    // reference: tensorflow/lite/kernels/internal/reference/reference_ops.h
-    int32_t shift = 10;
-
-    // 1.0 is equivalent to (1 << shift) in quantized space.
-    // Here we noted as unit = (1 << shift).
-    double unit = static_cast<double>(1 << shift);
-
-    // Stride and Offset is int16.
-    int32_t stride_y = std::lround(fp_stride_y * unit);
-    int32_t stride_x = std::lround(fp_stride_x * unit);
-    int32_t offset_y = std::lround(fp_offset_y * unit);
-    int32_t offset_x = std::lround(fp_offset_x * unit);
-
-    // Numerically we can decrement shift to let these number fits within 16
-    // bits but that's not commonly seen and won't match TFLite reference
-    if (stride_y > std::numeric_limits<int16_t>::max() ||
-        stride_x > std::numeric_limits<int16_t>::max() ||
-        stride_y < std::numeric_limits<int16_t>::min() ||
-        stride_x < std::numeric_limits<int16_t>::min() ||
-        offset_y > std::numeric_limits<int16_t>::max() ||
-        offset_x > std::numeric_limits<int16_t>::max() ||
-        offset_y < std::numeric_limits<int16_t>::min() ||
-        offset_x < std::numeric_limits<int16_t>::min()) {
-      op->emitOpError("OpResize: stride or offset out of 16 bits");
+    // It isn't commonly seen these numbers aren't fit within 16 bits, and won't
+    // match TFLite reference.
+    if (!isInt16Range(scale_y_n) || !isInt16Range(scale_y_d) ||
+        !isInt16Range(scale_x_n) || !isInt16Range(scale_x_d) ||
+        !isInt16Range(offset_y) || !isInt16Range(offset_x) ||
+        !isInt16Range(border_y) || !isInt16Range(border_x)) {
+      op->emitOpError(
```

Comment:
Use rewriter notifyMatchFailure as this is failure of pattern/conversion but the input op doesn't have an error

Reply:
Got you. Done.